### PR TITLE
Refactor docs

### PR DESCRIPTION
**Pull request by jolheiser**
This was intended to be a small followup for https://github.com/go-gitea/gitea/pull/23712, but...here we are.

1. Our docs currently use `slug` as the entire URL, which makes refactoring tricky (see https://github.com/go-gitea/gitea/pull/23712). Instead, this PR attempts to make future refactoring easier by using slugs as an extension of the section. (Hugo terminology)
  - What the above boils down to is this PR attempts to use directory organization as URL management. e.g. `usage/comparison.en-us.md` -> `en-us/usage/comparison/`, `usage/packages/overview.en-us.md` -> `en-us/usage/packages/overview/`
  - Technically we could even remove `slug`, as Hugo defaults to using filename, however at least with this PR it means `slug` only needs to be the name for the **current file** rather than an entire URL
2. This PR adds appropriate aliases (redirects) for pages, so anything on the internet that links to our docs should hopefully not break.
3. A minor nit I've had for a while, renaming `seek-help` to `support`. It's a minor thing, but `seek-help` has a strange connotation to it.
4. The commits are split such that you can review the first which is the "actual" change, and the second is added redirects so that the first doesn't break links elsewhere.

---

I am not sure if this is worth it vs going all in on the docusaurus rewrite.

tl;dr this PR is the answer to https://github.com/go-gitea/gitea/pull/23712#issuecomment-1484106990 (and a more general cleanup as well), but perhaps we should just use the shim originally proposed by https://github.com/go-gitea/gitea/pull/23712/commits/7dea95b92d8c33db3f3e18467e6a24cda48869c9 and turn focus to the other docs.

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
unfortunately a lot of conflicts.

**Comment by lunny** [comment]
This PR needs to be merged ASAP.

**Review by lunny — APPROVED** [review]
